components/script/dom/servoparser/async_html.rs

```diff
@@ -344,21 +344,23 @@ impl Tokenizer {
         &self.url
     }
 
-    pub fn set_plaintext_state(&mut self) {
+    pub fn set_plaintext_state(&self) {
         self.html_tokenizer_sender
             .send(ToHtmlTokenizerMsg::SetPlainTextState)
             .unwrap();
     }
 
-    fn insert_node(&mut self, id: ParseNodeId, node: Dom<Node>) {
-        assert!(self.nodes.insert(id, node).is_none());
+    fn insert_node(&self, id: ParseNodeId, node: Dom<Node>) {
+        assert!(self.nodes.borrow_mut().insert(id, node).is_none());
     }
 
-    fn get_node<'a>(&'a self, id: &ParseNodeId) -> &'a Dom<Node> {
-        self.nodes.get(id).expect("Node not found!")
+    fn get_node<'a>(&'a self, id: &ParseNodeId) -> Ref<'a, Dom<Node>> {
```

Comment:
I spent a bit of time looking at the re-entrant safety of this in https://github.com/servo/servo/blob/843726cdf580be6453a0b93f9a3adf0ca071deb2/components/script/dom/servoparser/async_html.rs#L403, and while it seems safe I don't like it. Should we change this to return a domroot? In many cases the dom is turned into one at the point of use anyway. Seems like a risky and probably useless optimization.

Reply:
I don't remember the precise history of why we used Dom as the handle type, but since this is preexisting code I'll file a followup issue.

Reply:
#33011